src/SAWScript/Crucible/MIR/Setup/Value.hs

```diff
@@ -60,7 +60,7 @@ import qualified SAWScript.Crucible.Common.Setup.Value as MS
 
 type instance MS.XSetupNull MIR = Void
 type instance MS.XSetupGlobal MIR = Void
-type instance MS.XSetupStruct MIR = ()
+type instance MS.XSetupStruct MIR = M.Adt
```

Comment:
A key change in this PR: we now bundle an `Adt` along with a MIR `SetupStruct`. This is used in several parts of the patch to consult the `Adt` definition of a struct (e.g., what the types of its fields are, whether it is a `repr(transparent)` struct or not, etc.)